src/runway/commands/modules_command.py

```diff
@@ -322,6 +331,65 @@ def validate_account_credentials(deployment, context):
         validate_account_alias(boto3.client('iam', **boto_args),
                                account_alias)
 
+class SourceProcessor(object):
```

Comment:
should we treat this as a `Path` class rather than just a source processor? We could do more here like moving the logic that exists later to ensure the path is a dir and determining the module type since it partially pertains to the provided path.

```
class Path(object):
	"""Metadata and methods for a module path."""

	def __init__(self, path, cache_dir=None):
        # type: (str, Optional[str]) -> None:
        """Initialize class.

        Args:
            path: The path to a module provided in the runway config.
            cache_dir: Optionally supply a preferred cache directory.

        """
        self._path = path  # private, unresolved path
        self.cache_dir = cache_dir or '.runway'

    @property
    def module_type(self):
        """Logic to determine the module type from the path provided."""

    @property
    def source_type(self):
        """Local, git, etc."""

    @property
    def uri(self):
        """Parsing for git uri."""

    def resolve(self):
        """Returns the resolved path and fetches if needed."""
```

Going with this approach, it would also be logical to split to its own file. I would also consider consuming this in `config.ModuleDefinition` if we wait to `resolve()` here.

Thoughts?

Reply:
I like that idea, I especially like the idea of getting this moved to it's own separate file here eventually. These classes are getting quite large and could get unruly if we keep adding.

I'll keep plugging away until I have a working model and then we'll start working in the refactors.

Reply:
Latest refactors have a lot more abstractions. Take a peek and let me know if there's other things we could be doing better.